maxim/logger/smolagents/client.py

```
@@ -0,0 +1,539 @@
+        gen_ctx_token = _current_generation_id.set(generation_id)
+        
+        # Attach trace to tools for instrumentation
+        attach_trace_to_tools(self, trace)
+        
+        try:
+            # Call original method
+            result = original_method(self, task, *args, **kwargs)
+            
+            # Handle streaming/generator results
+            if inspect.isgenerator(result):
+                def _iterate():
```

Ruff (0.12.2)

Missing return type annotation for private function `_iterate`

(ANN202)